packages/cli/create-strapi-starter/utils/build-starter.js

```diff
@@ -88,28 +88,35 @@ async function installWithLogs(path) {
 /**
  * @param  {Object} projectArgs - The arguments for create a project
  * @param {string|null} projectArgs.projectName - The name/path of project
- * @param {string|null} projectArgs.starterUrl - The GitHub repo of the starter
+ * @param {string|null} projectArgs.starter - The npm package of the starter
  * @param  {Object} program - Commands for generating new application
  */
-module.exports = async function buildStarter(programArgs, program) {
-  let { projectName, starterUrl } = programArgs;
-
-  // Fetch repo info
-  const repoInfo = await getRepoInfo(starterUrl);
-  const { fullName } = repoInfo;
-
-  // Create temporary directory for starter
-  const tmpDir = await fse.mkdtemp(join(os.tmpdir(), 'strapi-'));
-
-  // Download repo inside temporary directory
-  await downloadGitHubRepo(repoInfo, tmpDir);
-
-  const starterJson = readStarterJson(join(tmpDir, 'starter.json'), starterUrl);
+module.exports = async function buildStarter({ projectName, starter }, program) {
+  let starterPath;
+  let starterParentPath;
+  let starterPackageInfo = {};
+  const isLocalStarter = ['./', '../', '/'].some(filePrefix => starter.startsWith(filePrefix));
 
   // Project directory
   const rootPath = resolve(projectName);
   const projectBasename = basename(rootPath);
 
+  if (isLocalStarter) {
+    // Starter is a local directory
+    console.log('Installing local starter.');
+    starterPath = resolve(rootPath, '..', starter);
+  } else {
+    // Starter should be an npm package. Fetch starter info
+    starterPackageInfo = await getStarterPackageInfo(starter);
+    console.log(`Installing ${chalk.yellow(starterPackageInfo.name)} starter.`);
+
+    // Download starter repository to a temporary directory
+    starterParentPath = await fse.mkdtemp(join(os.tmpdir(), 'strapi-'));
+    starterPath = await downloadNpmStarter(starterPackageInfo, starterParentPath);
+  }
```

Comment:
I'm thinking this could maybe be a function outside of `buildStarter`
```
const getStarterInfo = (starter) => {
  let starterPath;
  let starterParentPath;
  let starterPackageInfo = {};
  // ...do stuff
  return { starterPath, starterParentPath, starterPackageInfo }
}

```
That way the mutable variables are scoped and you can destructure constants.  
```
const { starterPath, starterParentPath, starterPackageInfo } = getStarterInfo(starter)
```

Reply:
Good idea :)